tests/domains_test.go

```diff
@@ -1,21 +1,134 @@
 package tests
 
 import (
+	"fmt"
+	"math/rand"
+	"os"
+	"strconv"
+
 	. "github.com/onsi/ginkgo"
-	// . "github.com/onsi/gomega"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+	. "github.com/onsi/gomega/gexec"
 )
 
+func getRandDomain() string {
+	return fmt.Sprintf("my-custom-%d.domain.com", rand.Intn(999999999))
+}
+
 var _ = Describe("Domains", func() {
+	var testApp App
+	var domain string
+
+	Context("with app yet to be deployed", func() {
+
+		BeforeEach(func() {
+			domain = getRandDomain()
+			// TODO: gitInit()
+			cmd, err := start("git init")
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(cmd).Should(Say("Initialized empty Git repository"))
+
+			testApp.Name = getRandAppName()
+			cmd = createApp(testApp.Name)
+			Eventually(cmd).Should(SatisfyAll(
+				Say("Git remote deis added"),
+				Say("remote available at ")))
+		})
+
+		AfterEach(func() {
+			destroyApp(testApp)
+			// TODO: gitClean()
+			cmd, err := start("rm -rf .git")
```

Comment:
Yes, that would also work.  I just pulled over the `rm -rf .git` block from another test and wrote a TODO more as a reminder to transfer that and the `git init` logic into re-usable utils (to be placed in `tests_suite_test` for now, since we don't have a dedicated `utils` resource as of yet.)  How does that sound?  Then any test could simply call `gitInit()` in setup and `gitClean()` in teardown